internal/pkg/cmd/prerunner.go

```diff
@@ -165,7 +165,11 @@ func (c *AuthenticatedCLICommand) AuthToken() string {
 	return c.Context.GetAuthToken()
 }
 
-func (c *AuthenticatedCLICommand) EnvironmentId() string {
+func (c *AuthenticatedCLICommand) EnvironmentId(command *cobra.Command) string {
```

Comment:
Instead of passing `cmd` everywhere (this makes functions significantly harder to unit test!) can we return the environment ID and the suggestion (which may be "") in a similar manner to how we return an error in Go?

Reply:
Then we'll need to add `if warning!="", printf warning` everywhere `c.EnvironmentId()` is called. That's less efficient and more duplicate code

Reply:
We could go for the middle ground: `_, _ = fmt.Fprint(os.Stderr, errors.EnvNotSetErrorMsg+noEnvSuggestions+"\n")` so that we don't need to pass in command just for the `ErrPrint` and we don't need to check for the empty string at every call.

Reply:
Hmm, good call @sgagniere. I think we used to pass `cmd` everywhere so our unit tests (which were really integration tests) could record the output. We could make a PR to rewrite the `util.Print*()` functions so they don't need `cmd` (and during that process, rewrite the unit tests that break into integration tests).

Reply:
^ If the above sounds good, let's merge this PR as-is.

Reply:
Sounds good to me too.

Reply:
I'll take care of this today!